enterprise_access/apps/api/serializers/content_assignments/assignment.py

```diff
@@ -100,10 +110,34 @@ class LearnerContentAssignmentAdminResponseSerializer(LearnerContentAssignmentRe
         ),
         choices=AssignmentLearnerStates.CHOICES,
     )
+    error_reason = serializers.SerializerMethodField()
 
     class Meta(LearnerContentAssignmentResponseSerializer.Meta):
         fields = LearnerContentAssignmentResponseSerializer.Meta.fields + [
             'recent_action',
             'learner_state',
+            'error_reason',
         ]
         read_only_fields = fields
+
+    def get_error_reason(self, assignment):
```

Comment:
In order for this field to show up in redoc/openapi as the correct type, you may need to decorate this method with

```python
@extend_schema_field(serializers.CharField)
```